src/cr/cube/matrix/measure.py

```diff
@@ -1320,6 +1325,118 @@ def blocks(self):
         )
 
 
+# === MARGINALS ===
+
+
+class _BaseSecondOrderMarginal(object):
+    """Base class for all second-order measure objects."""
+
+    def __init__(self, dimensions, second_order_measures, cube_measures):
+        self._dimensions = dimensions
+        self._second_order_measures = second_order_measures
+        self._cube_measures = cube_measures
+
+    @lazyproperty
+    def blocks(self):
+        """List of the 2 1D ndarray "blocks" making up this measure.
+
+        These are the base-values and the subtotals.
+        """
+        raise NotImplementedError(  # pragma: no cover
+            "%s must implement `.blocks`" % type(self).__name__
+        )
+
+    @lazyproperty
+    def direction(self):
+        """String, either "row" or "column", the directionality the marginal."""
+        raise NotImplementedError(  # pragma: no cover
+            "%s must implement `.direction`" % type(self).__name__
+        )
+
+    @lazyproperty
```

Comment:
It occurs to me that this (and another one below) don't really need to be lazyproperties, they could just be properties. Should I change?